app/javascript/dashboard/helper/clone.js

```diff
@@ -0,0 +1,33 @@
+/**
+ * Creates a deep clone of the provided object.
+ *
+ * This function attempts to use the `structuredClone` method if available.
+ * If `structuredClone` is not supported, it falls back to using
+ * `JSON.parse(JSON.stringify())`.
+ *
+ * @param {*} obj - The object to be cloned. Can be of any type.
+ * @returns {*} A deep clone of the input object.
+ *
+ * @throws {TypeError} If the object contains values that JSON cannot serialize
+ *                     (e.g., functions, undefined) when falling back to
+ *                     JSON methods.
+ *
+ * @example
+ * const original = { a: 1, b: { c: 2 } };
+ * const clone = cloneObject(original);
+ * console.log(clone); // { a: 1, b: { c: 2 } }
+ * console.log(original === clone); // false
+ */
+export function cloneObject(obj) {
+  if (typeof structuredClone === 'function') {
```

Comment:
I would suggest adding a try-catch block could make the function more robust.

```
try {
    return structuredClone(obj);
  } catch (error) {
    try {
      return JSON.parse(JSON.stringify(obj));
    } catch (error) {
      throw new TypeError('Failed to clone object.');
    }
}
```

Reply:
An error will be thrown anyway, seems futile to catch the error and then throw another one while handling that.